keras_nlp/layers/modeling/cached_multi_head_attention.py

```diff
@@ -87,8 +85,12 @@ def call(
         cache=None,
         cache_update_index=None,
     ):
-        if not self._built_from_signature:
+        if (
+            hasattr(self, "_build_from_signature")
```

Comment:
Did we change the API?

Reply:
Short version, this is a gross bit of `tf.keras` we don't need to deal with in `keras-core`.

Long version, `tf.keras` has issues supporting multiple arguments to call/build, resulting in the `MultiHeadAttention` layer working around the normal `build` function with `_build_from_signature` and hitting some issues with serialization. This was not a hack that got propogated to keras-core thankfully, normal `build` can now support multiple build arguments.

Can post a more in the weeds explainer if you would like

Reply:
Yeah I've seen some bugs around that one. Good fix